Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp

```diff
@@ -2184,9 +2184,26 @@ ExceptionOr<RefPtr<CanvasPattern>> CanvasRenderingContext2DBase::createPattern(C
     return Exception { TypeError };
 }
 
-void CanvasRenderingContext2DBase::didDrawEntireCanvas()
+OptionSet<CanvasRenderingContext2DBase::DidDrawOption> CanvasRenderingContext2DBase::defaultDidDrawOptions()
 {
-    didDraw(backingStoreBounds(), { DidDrawOption::ApplyClip, DidDrawOption::ApplyPostProcessing });
+    static OptionSet<DidDrawOption> defaultOptions { DidDrawOption::ApplyTransform, DidDrawOption::ApplyShadow, DidDrawOption::ApplyClip, DidDrawOption::ApplyPostProcessing };
+    return defaultOptions;
+}
+
+OptionSet<CanvasRenderingContext2DBase::DidDrawOption> CanvasRenderingContext2DBase::defaultDidDrawOptionsWithoutPostProcessing()
```

Comment:
This can be 

```
static OptionSet<DidDrawOption>  defaultDidDrawOptionsNoPostProcessing() 
{
   return  { DidDrawOption::ApplyTransform, DidDrawOption::ApplyShadow, DidDrawOption::ApplyClip };
}
```

Reply:
Same reasoning as`defaultDidDrawOptions`.